src/wasm_opt.rs

```diff
@@ -0,0 +1,82 @@
+use anyhow::{ensure, Context, Result};
+use std::{path::PathBuf, process};
+
+#[allow(unreachable_code)]
+pub fn run(
+    binary: Vec<u8>,
+    shrink_level: u32,
+    optimization_level: u32,
+    debug_info: bool,
+) -> Result<Vec<u8>> {
```

Comment:
This API can be improved.

Drop the binary field in favor of a file path. Then the result should be `Result<()>`.

Make the optimization in place. You won't need tempfile anymore.